plugins/cpp/parser/src/symbolhelper.cpp

```diff
@@ -69,12 +74,25 @@ std::string getMangledName(
   const clang::NamedDecl* nd_,
   const model::FileLoc& fileLoc_)
 {
+
   if (const clang::VarDecl* vd = llvm::dyn_cast<clang::VarDecl>(nd_))
   {
     std::string result = nd_->getQualifiedNameAsString();
 
     if (const clang::DeclContext* dc = nd_->getParentFunctionOrMethod())
     {
+      // there are contexts which getParentFunctionOrMethod() is not null
+      // but the result is not FunctionDecl
+      // like CapturedDecl	    
+      if ( 0 == llvm::dyn_cast<clang::FunctionDecl>(dc) )
```

Comment:
In addition, `dyn_cast` returns a pointer, so the convention is using `if (const auto* P = dyn_cast<...>(...))`.

If you don't need the returned cast pointer, you could also write:

`if (!llvm::isa<T>(obj))`

which returns a `bool`.